src/coprocessor/dag/expr/ctx.rs

```diff
@@ -70,60 +69,112 @@ pub struct EvalConfig {
 
 impl Default for EvalConfig {
     fn default() -> EvalConfig {
-        EvalConfig::new(0).unwrap()
+        EvalConfig::new()
     }
 }
 
 impl EvalConfig {
-    pub fn new(flags: u64) -> Result<EvalConfig> {
-        let e = EvalConfig {
+    pub fn new() -> EvalConfig {
+        EvalConfig {
             tz: Tz::utc(),
-            ignore_truncate: (flags & FLAG_IGNORE_TRUNCATE) > 0,
-            truncate_as_warning: (flags & FLAG_TRUNCATE_AS_WARNING) > 0,
-            overflow_as_warning: (flags & FLAG_OVERFLOW_AS_WARNING) > 0,
-            in_insert_stmt: (flags & FLAG_IN_INSERT_STMT) > 0,
-            in_update_or_delete_stmt: (flags & FLAG_IN_UPDATE_OR_DELETE_STMT) > 0,
-            in_select_stmt: (flags & FLAG_IN_SELECT_STMT) > 0,
-            pad_char_to_full_length: (flags & FLAG_PAD_CHAR_TO_FULL_LENGTH) > 0,
-            divided_by_zero_as_warning: (flags & FLAG_DIVIDED_BY_ZERO_AS_WARNING) > 0,
+            ignore_truncate: false,
+            truncate_as_warning: false,
+            overflow_as_warning: false,
+            in_insert_stmt: false,
+            in_update_or_delete_stmt: false,
+            in_select_stmt: false,
+            pad_char_to_full_length: false,
+            divided_by_zero_as_warning: false,
             max_warning_cnt: DEFAULT_MAX_WARNING_CNT,
             sql_mode: 0,
             strict_sql_mode: false,
-        };
+        }
+    }
+
+    pub fn set_ignore_truncate(&mut self, new_value: bool) -> &mut Self {
```

Comment:
Please mark with `#[cfg(test)]` for those functions only used in the test.

Reply:
Are you sure? All the setters or just this?